docs/4.0-migration-guide/platform-entropy-gathering.md

```diff
@@ -0,0 +1,63 @@
+## Platform-specific Entropy Gathering
+
+### Disclaimer
+
+All the following sections refer to the case where entropy module uses default
+entropy sources, i.e. `MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES` is not defined.
+
+### Background
+
+Up to Mbed TLS 3.6 entropy module supported 3 entropy sources by default:
+
+- Platform
+  - Enabled by default it could be disabled through `MBEDTLS_NO_PLATFORM_ENTROPY`.
+  - It used platform-specific sources such as getrandom(), /dev/urandom or
+    BCryptGenRandom() to gather entropy data.
+  - Only useful when running on high level OSes, not in baremetal projects.
+
+- Hardware
+  - Disabled by default, it could be enabled through `MBEDTLS_ENTROPY_HARDWARE_ALT`.
+  - It allowed the user to define a custom function named `mbedtls_hardware_poll()`
+    to gather entropy data from some hardware source.
+  - Very useful for baremetal projects.
+  - Unfortunately the prototype for the function was in a private header file.
+
+- NV seed
```

Comment:
The NV seed part hasn't changed. It would be useful to mention it in passing, but the document should not suggest that it has changed or that it has been removed.

Reply:
That's part for the "Background" section indeed where I was trying to give the user an overview of the situation before my changes. I will add a note to clearly state that nothing has changed for NV seed